source/traction_boundary_conditions/lithostatic_pressure.cc

```diff
@@ -0,0 +1,469 @@
+          in.temperature[0] = this->get_initial_conditions().initial_temperature(in.position[0]);
+          // the previous pressure
+          in.pressure[0] = pressure[i-1];
+
+          // we need the actual composition at t0
+          for (unsigned int c=0; c<n_compositional_fields; ++c)
+            in.composition[0][c] = this->get_compositional_initial_conditions().initial_composition(in.position[0], c);
+
+          in.strain_rate.resize(0); // we do not need the viscosity
+          this->get_material_model().evaluate(in, out);
+
+          // get the magnitude of gravity. we assume
+          // that gravity always points along the depth direction. this
+          // may not strictly be true always but is likely a good enough
+          // approximation here.
+          const double density = out.densities[0];
+          const double gravity = this->get_gravity_model().gravity_vector(in.position[0]).norm();
+
+          // trapezoid integration
+          pressure[i] = sum + delta_z * 0.5 * density * gravity;
+          sum += delta_z * density * gravity;
+
+        }
+
+      Assert (*std::min_element (pressure.begin(), pressure.end()) >=
+              -std::numeric_limits<double>::epsilon() * pressure.size(),
+              ExcInternalError());
+
+    }
+
+    template <int dim>
+    Tensor<1,dim>
+    LP<dim>::
+    traction (const Point<dim> &p,
+              const Tensor<1,dim> &normal) const
+    {
+      // We want to set the normal component to the vertical boundary
+      // to the lithostatic pressure, the rest of the traction
+      // components are left set to zero
+      Tensor<1,dim> traction;
+
+      // assign correct value to traction
+      // get the lithostatic pressure from a linear interpolation of
+      // the calculated profile
+      traction = -get_pressure(p) * normal;
+      return traction;
+    }
+
+    template <int dim>
+    double
+    LP<dim>::
+    get_pressure (const Point<dim> &p) const
+    {
+      // depth at which we need the pressure
+      const double z = this->get_geometry_model().depth(p);
+
+      // TODO: probably this check is unnecessary as
+      // geometry models cut off depth at max_depth
+      if (z >= this->get_geometry_model().maximal_depth())
+        {
+          Assert (z <= this->get_geometry_model().maximal_depth() + delta_z,
+                  ExcInternalError());
+          // return deepest (last) pressure
+          return pressure.back();
+        }
+
+      const unsigned int i = static_cast<unsigned int>(z/delta_z);
+      Assert ((z/delta_z) >= 0, ExcInternalError());
+      Assert (i+1 < pressure.size(), ExcInternalError());
+
+      // now do the linear interpolation
+      const double d=1.0+i-z/delta_z;
+      Assert ((d>=0) && (d<=1), ExcInternalError());
+
+      return d*pressure[i]+(1-d)*pressure[i+1];
+    }
+
+    template <int dim>
+    Point<dim>
+    LP<dim>::
+    spherical_to_cart(const Point<dim >sphere_coord) const
```

Comment:
As you already noticed, the function `cartesian_coordinates` in utilities.cc does exactly the same, so you can just use that one.